frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -16,6 +17,115 @@ local T = require("ffi/util").template
 -- so we only have to look for them on the first :init()
 local available_ifos = nil
 
+-- largely thanks to https://tuxor1337.github.io/firedict/dictionaries.html
+local downloadable_dicts = {
```

Comment:
May be this could be imported from another file, for easier update by people?
And may be a LanguageSorter class that would make a first level menu my language, with the available dicts that have it as lang_in or lang_out :)

Reply:
The Czech dicts would get a double entry in the list? The zip includes both ways.

Reply:
They would appear only once in each of the English> and Czech> menu.

Reply:
That does seem like a good idea to sort them.

So drop the KeyValuePage idea alltogether and switch to individual MultiConfirmBoxes or something? Or per language KeyValuePage with slightly more generous info?

Reply:
Oh, right, this was KeyValuePage. I've been to much into Menu these last weeks :)
You could just add the languages as menu items, and they will launch a KeyValuePage just like you have there, with all the dicts related to this language.
Most users (and me) would probably only go see the dicts related to their mothertongue, and probably have a look at Latin and Greek (was ok to me that they translate to english and not to french).